### PR TITLE
feat: Initial implementation

### PR DESCRIPTION
**Pull request by andrefgneves**

### CODE REVIEW

**Comment by satazor** [comment]
We need webpack-sane-compiler-notifier first.. though it feels kind of strange.

What if we remove the reporter from this package and use it from the outside?

**Comment by andrefgneves** [comment]
Sounds good.

**Comment by andrefgneves** [comment]
I removed all reporting related functionality.

File: README.md [hunk 1]
```diff
@@ -1 +1,167 @@
...
+});
+```
+
+### .unwatch()
+
+Stops watching for changes.   
+Returns a promise that fulfills when done.
+
+
+### .resolve()
```
Comment (acostalima) [inline README.md]: `.resolve()` is not implemented in the source

File: test/unwatch.spec.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    it('should return a promise', () => {
+        const compiler = createCompiler(configBasic);
+
+        compiler.watch();
+
+        const promise = compiler.unwatch();
+
+        expect(promise).toBeDefined();
+        expect(typeof promise.then).toBe('function');
```
Comment (acostalima) [inline test/unwatch.spec.js]: The test will pass if `promise` is an object with a `then` function, right? Don't we want to ensure that we get an instance of a `Promise`?

Comment (andrefgneves) [reply test/unwatch.spec.js]: Done

File: README.md (hunk 1 above)
Comment (andrefgneves) [reply README.md]: Done

File: package.json [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  "description": "A webpack compiler wrapper that adds sanity and pretty reporting",
+  "keywords": [
+    "webpack",
+    "compiler",
+    "compilation",
+    "report",
+    "reporter",
+    "reporting"
+  ],
+  "author": "André Cruz <andre@moxy.studio>",
```
[the remark this answers is not in the document]
Comment (andrefgneves) [reply package.json]: Done

**Pull request merged**